### PR TITLE
Raise an error for negative sigma values in guassian_filter.

### LINKED ISSUE #1586: Document negative `sigma` values in `filters.gaussian_filter` are clipped to zero.
Negative sigma values have no effect; they are clipped to zero. This should be documented.

### PR DESCRIPTION
**Pull request by jwiggins**
Closes #1586

### CODE REVIEW

File: skimage/filters/_gaussian.py [hunk 1]
```diff
@@ -93,6 +93,9 @@ def gaussian_filter(image, sigma, output=None, mode='nearest', cval=0,
                "3D image with last dimension of length 3.")
         warnings.warn(RuntimeWarning(msg))
         multichannel = True
+    if np.any(np.asarray(sigma) < 0.0):
+        msg = "Sigma values less than zero are not valid"
```
Comment (stefanv) [inline skimage/filters/_gaussian.py]: You can combine these two lines: ``raise ValueError("Sigma values...")``

**Comment by stefanv** [comment]
Apart from a minor style preference, this looks good to me.

File: skimage/filters/_gaussian.py (hunk 1 above)
Comment (jwiggins) [inline skimage/filters/_gaussian.py]: Right... I was expecting that to be a longer message when I started writing it. Done.

File: skimage/filters/tests/test_gaussian.py [hunk 2]
```diff
@@ -1,8 +1,23 @@
+from nose.tools import raises
 import numpy as np
 from skimage.filters._gaussian import gaussian_filter
 from skimage._shared._warnings import expected_warnings
 
 
+@raises(ValueError)
```
Comment (emmanuelle) [inline skimage/filters/tests/test_gaussian.py]: could use ``numpy.testing.assert_raises`` instead ?

Comment (jwiggins) [inline skimage/filters/tests/test_gaussian.py]: Yup. It'll make for fewer tests too

**Comment by emmanuelle** [comment]
Looks great to me. I'm just waiting for CI to finish (which might take longer than the sprint :-)), and then I'll merge since @stefanv also likes it.

**Comment by emmanuelle** [comment]
Merged, thank you!

**Pull request merged**